js/krakenfu.js

```diff
@@ -0,0 +1,1196 @@
+    }
+
+    parseSymbolIdJoined (symbolId) {
+        // Convert by detecting and converting currencies in symbol
+        const symbolIdLower = symbolId.toLowerCase ();
+        const quoteIds = this.options['symbol']['quoteIds'];
+        const reversed = this.options['symbol']['reversed'];
+        const method = reversed ? 'startsWith' : 'endsWith';
+        let quoteId = undefined;
+        let baseId = undefined;
+        for (let i = 0; i < quoteIds.length; i++) {
+            if (this[method] (symbolIdLower, quoteIds[i].toLowerCase ())) {
+                quoteId = quoteIds[i];
+                break;
+            }
+        }
+        if (quoteId === undefined) {
+            throw new BadSymbol (this.id + ' symbolId could not be parsed: ' + symbolId);
+        }
+        if (!reversed) {
+            const baseIdLength = symbolId.length - quoteId.length;
+            baseId = this.sliceString (symbolId, 0, baseIdLength);
+            quoteId = this.sliceString (symbolId, baseIdLength);
+        } else {
+            quoteId = this.sliceString (symbolId, 0, quoteId.length);
+            baseId = this.sliceString (symbolId, quoteId.length);
+        }
+        return {
+            'baseId': baseId,
+            'quoteId': quoteId,
+            'base': this.safeCurrencyCode (baseId),
+            'quote': this.safeCurrencyCode (quoteId),
+        };
+    }
+
+    startsWith (string, x) {
+        return this.sliceString (string, 0, x.length) === x;
+    }
+
+    endsWith (string, x) {
+        const start = Math.max (0, string.length - x.length);
+        return this.sliceString (string, start) === x;
+    }
+
+    sliceString (string, start = undefined, end = undefined) {
+        if (start === undefined) {
+            start = 0;
+        }
+        if (end === undefined) {
+            end = string.length;
+        }
+        return string.slice (start, end);
+    }
+
+    nonce () {
+        return this.milliseconds ().toString ();
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        const endpoint = '/api/' + this.version + '/' + path;
+        let query = endpoint;
+        let postData = '';
+        if (Object.keys (params).length) {
+            postData = this.urlencode (params);
+            query += '?' + postData;
+        }
+        const url = this.urls['api'][api] + query;
+        if (api === 'private') {
+            const nonce = ''; // this.nonce ();
+            const auth = postData + nonce + endpoint; // 1
+            const hash = this.hash (this.encode (auth), 'sha256', 'binary'); // 2
+            const secret = this.base64ToBinary (this.secret); // 3
+            const signature = this.hmac (hash, secret, 'sha512', 'base64'); // 4-5
+            headers = {
+                'Content-Type': 'application/json',
+                'APIKey': this.apiKey,
+                'Authent': this.decode (signature),
```

Comment:
in the python version from your awesome unmerged library, thanks! I'm getting the error:
AttributeError: 'str' object has no attribute 'decode'
changing this to 
'Authent': signature, 

seems to fix the problem

Reply:
@qtmspin thanks, this was due to ccxt upgrade. It is now fixed in ccxt-unmerged, gonna make changes in this PR later.